docs/src/main/docs/guides/91_mp-tutorial.adoc

```diff
@@ -0,0 +1,1267 @@
+ MicroProfile 2.2 features in the application.
+
+TIP: MicroProfile 2.2 contains features like Metrics, Health Check,
+ Streams Operators, Open Tracing, OpenAPI, REST client, and fault
+ tolerance. You can find detailed information about MicroProfile 2.2 on the
+ https://projects.eclipse.org/projects/technology.microprofile/releases/microprofile-2.2[Eclipse MicroProfile 2.2] site.
+
+With this POM file, the application can be built successfully with Maven:
+
+[source,bash]
+----
+mvn clean package
+----
+
+This will create a JAR file in the `target` directory.
+
+TIP: The warning message `JAR will be empty - no content was marked for inclusion!` 
+ can be ignored for now  because there is no actual content in the 
+ application yet.
+
+== Create the MicroProfile application
+
+The actual application logic can be created now.
+ The first step is to add the application class. Create a directory
+ for your source code, and then create directories for the package 
+ hierarchy:
+
+[source,bash]
+.Create directories for source code
+----
+mkdir -p src/main/java/io/helidon/examples
+----
+
+The application will be a simple REST service that will return a 
+ greeting to the caller.  The first iteration of the application will 
+ contain a MicroProfile application class, a resource class and a 
+ Main class which will be used to start up the Helidon server 
+ and the application.
+
+Create a new Java class inside that directory called `GreetApplication.java`
+ and add the following content:
+
+[source,java]
+.Initial application class
+----
+package io.helidon.examples;
+
+import io.helidon.common.CollectionsHelper;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.ApplicationPath;
+import javax.ws.rs.core.Application;
+import java.util.Set;
+
+@ApplicationScoped <1>
+@ApplicationPath("/") <2>
+public class GreetApplication extends Application { <3>
+
+    @Override 
+    public Set<Class<?>> getClasses() { <4>
+        return CollectionsHelper.setOf(GreetResource.class);
+    }
+
+}
+----
+
+
+<1> The class is annotated with `ApplicationScoped` 
+ which specifies that this bean is "application scoped". The application scope
+ is shared between all web service invocations that execute within the same
+ application. 
+<2> The `ApplicationPath` annotation sets the URL path that will be used to 
+ access the application.
+<3> The application class must extend `javax.ws.rs.core.Application`. 
+<4> In the application class, override the 
+ `public Set<Class<?>> getClasses()` method.  This method must return
+ a `Set` of `Class` objects, one for each resource that the application 
+ will serve.  Currently, the application only has one resource, the
+ `GreetResource`.
+
+TIP:  You can learn more about scopes and contexts; and how they are used from the https://javaee.github.io/javaee-spec/javadocs/javax/enterprise/context/package-summary.html[Java EE
+ Specification].
```

Comment:
Prefer https://docs.jboss.org/cdi/api/2.0/index.html, the official specification javadocs; this also helps point out that we aren't connected with Java EE 8.